src/styles/main.scss

```diff
@@ -0,0 +1,54 @@
+body {
+  margin: 0;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
+  font-family: Arial, Helvetica, sans-serif;
+}
+
+.calendar {
+  width: $calendar-size;
+  display: flex;
+  padding: $padding;
+  justify-content: left;
+  flex-wrap: wrap;
+  gap: $block-gap;
+  font-size: 30px;
+
+  @each $day, $offset in $days-of-week {
+    &--start-day-#{$day} &__day:first-child {
+      margin-left: calc(#{$offset} * ($block-size + $block-gap));
+    }
+  }
+
+  @for $i from 28 through 31 {
+    &--month-length-#{$i} &__day:nth-child(n + #{$i + 1}) {
+      display: none;
```

Comment:
The loop using '@for' is used to control the visibility of '.calendar__day' elements based on the length of the month. Validate that the logic aligns with the requirements of the feature it's intended for.